src/compiletest.md

```diff
@@ -0,0 +1,184 @@
+# `compiletest`
+## Introduction
+`compiletest` is the main test harness of the Rust test suite.  It allows test authors to organize large numbers of tests (the
+Rust compiler has many thousands), efficient test execution (parallel execution is supported), and allows the test author to
+configure behavior and expected results of both individual and groups of tests.
+
+`compiletest` tests may check test code for success, for failure or in some cases, even failure to compile.  Tests are
+typically organized as a Rust source file with annotations in comments before and/or within the test code, which serve to
+direct `compiletest` on if or how to run the test, what behavior to expect, and more.  If you are unfamiliar with the compiler
+testing framework, see [`this chapter`](https://github.com/rust-lang-nursery/rustc-guide/blob/master/src/tests/intro.md) for additional background.
+
+The tests themselves are typically (but not always) organized into "suites"--for example, `run-pass`, a folder
+representing tests that should succeed, `run-fail`, a folder holding tests that should compile successfully, but return
+a failure (non-zero status), `compile-fail`, a folder holding tests that should fail to compile, and many more.  The various
+suites are defined in [src/tools/compiletest/src/common.rs](https://github.com/rust-lang/rust/tree/master/src/tools/compiletest/src/common.rs) in the `pub struct Config` declaration.  And a very good 
+introduction to the different suites of compiler tests along with details about them can be found
+[here, at Brian Anderson's blog](https://brson.github.io/2017/07/10/how-rust-is-tested#s-ct).
```

Comment:
Seems a bit weird to me to provide the same link that we provide in the next paragraph...  Plus, we're not even talking about adding tests in this paragraph.

LMK if you'd still like to link to Adding tests here, and if you'd like it alongside or instead of the link to @brson's blog.

Reply:
As I think about it more, I suppose it's also weird to provide two different authoritative sources on a topic...  ;)  I think I'll just replace the brson link.  LMK if you have other thoughts...

Reply:
I think there sounds reasonable. There is already a link to brsons post in the other chapter, so let's just point to the other chapter everywhere.